javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithAnnotations.java

```diff
@@ -60,27 +60,53 @@ default N addAnnotation(AnnotationExpr element) {
      * Annotates this
      *
      * @param name the name of the annotation
+     * @return this
+     */
+    @SuppressWarnings("unchecked")
+    default N addAnnotation(String name) {
+        NormalAnnotationExpr annotation = new NormalAnnotationExpr(
+                Name.parse(name), new NodeList<>());
+        getAnnotations().add(annotation);
+        return (N) this;
+    }
+
+    /**
+     * Annotates this
+     *
+     * @param name the name of the annotation
      * @return the {@link NormalAnnotationExpr} added
      */
-    default NormalAnnotationExpr addAnnotation(String name) {
-        NormalAnnotationExpr normalAnnotationExpr = new NormalAnnotationExpr(
+    @SuppressWarnings("unchecked")
+    default NormalAnnotationExpr addAndGetAnnotation(String name) {
+        NormalAnnotationExpr annotation = new NormalAnnotationExpr(
                 Name.parse(name), new NodeList<>());
-        getAnnotations().add(normalAnnotationExpr);
-        return normalAnnotationExpr;
+        getAnnotations().add(annotation);
+        return annotation;
     }
 
     /**
-     * Annotates this and automatically add the import
+     * Annotates this node and automatically add the import
      *
      * @param clazz the class of the annotation
-     * @return the {@link NormalAnnotationExpr} added
+     * @return this
      */
-    default NormalAnnotationExpr addAnnotation(Class<? extends Annotation> clazz) {
+    default N addAnnotation(Class<? extends Annotation> clazz) {
         ((Node) this).tryAddImportToParentCompilationUnit(clazz);
```

Comment:
This functionality wasn't changed (what I did was correct the use of the "add" and "addAndGet" pattern) but it is a good idea: #803